### ISSUE TITLE
[B2] Wrong USD equivalent on website&mobile

### ISSUE THREAD

**Issue by kerrilija** [asastats/channel#683]
By UltimateSideQuest in [Discord](https://discord.com/channels/906917846754418770/1028021510453084161/1124916501644841000):

> The displayed price for Algo seems to be stuck on .1214 for the duration of the day

![Screenshot_20230701-221523.png](https://github.com/asastats/channel/assets/82002056/19ae18e8-1fff-4d8d-a478-3eab07ab75bf)

![Screenshot_20230701-221452.png](https://github.com/asastats/channel/assets/82002056/59a274b5-ece7-43f4-be02-6ae3c3ae7249)

**Comment by ipaleka** [comment]
ASA Stats defines USD price simply as USDC unit price for ALGO on the Algorand blockchain, by evaluating 1 USDC against all the providers.

That means that USD price on ASA Stats will remain the same in period wwhen there's no activity that will lower down or raise up the highest price user can get for 1 USDC on currently the best AMM pool. Maybe on some pools there are a lot of activity, but such an activity doesn't cross the price we fetch from the best provider for **1 USDC**.

I checked a few times during writing this through our API (that brings enough places after decimal point) and here's the results:

```
curl --http1.1 --header "Content-Type: application/json" --header "Authorization: Bearer xxxxxxxx" -X GET
https://www.asastats.com/api/2EVGZ4BGOSL3J64UYDE2BUGTNTBZZZLI54VUQQNZZLYCDODLY33UGXNSIU/?objects=total

{..."pricealgo":0.12142812444267545,...}
{..."pricealgo":0.12142813375176441,...}
{..."pricealgo":0.12142813375176441,...}
{..."pricealgo":0.12137985383097095,...}
```

Here's a confirmation from a random account that our engine works as expected:

<img src="https://github.com/asastats/channel/assets/49662536/60ff165f-f43a-4f53-864d-17af7cea21ac" width="280">
<img src="https://github.com/asastats/channel/assets/49662536/65538fe5-e852-43a6-a662-eae96adb7762" width="280">

-----

Btw, here's the related discussion: https://github.com/asastats/channel/discussions/53

**Comment by ipaleka** [comment]
By [UltimateSideQuest](https://discord.com/channels/906917846754418770/1028021510453084161/1125086218883969106):
> Here's a confirmation from a random account that our engine works as expected:

Really? That's interesting because I've used the ASAStats app since it's inception, I check my portfolio numerous times each day and this is the first time I saw the displayed price of Algo unchanged for a whole day

So during all this price movement, there's not enough activity to lower or raise the highest price for 1 USD on the best AMM pool.
I wouldn't expect that, tbh but if that's what you say (edited)

<img src="https://github.com/asastats/channel/assets/49662536/e7abe299-a118-40ee-bad8-cddb3dc0bd1b" width="400">

-----

ipaleka — Today at 5:38 PM
I expect that the only one pool that provides such a value during the last day, let me check and I'll get back to you.

I have to admit that everything sounds the way you're presenting - wrong - tho I can't get grasp why because engine picks up the pools' values correctly. My educated guess is that engine in production somehow picks some proxy pool I can't locate on development machine.
